File: backend/albergue1/models/caracteristica_complementaria.py
from utils.db import db

class Caracteristica_complementaria(db.Model):
    __tablename__ = 'caracteristica_complementaria'

    # Variables
    id_caracteristica = db.Column(
        db.Integer, 
        primary_key=True,
        autoincrement=True,
        nullable=False
    )

    nombre = db.Column(
        db.String(15),
        nullable=False,
        unique=True
    )

    unidad_medida = db.Column(
        db.String(7),
        nullable=False
    )

    # Constructor
    def __init__(self,
                nombre,
                unidad_medida):
        self.nombre=nombre
        self.unidad_medida=unidad_medida

File: backend/albergue2/src/main/java/com/adoptame/albergue2/model/CaracteristicaComplementaria.java
package com.adoptame.albergue2.model;

import jakarta.persistence.*;

@Entity
@Table(name = "CARACTERISTICA_COMPLEMENTARIA")
public class CaracteristicaComplementaria {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_CARACTERISTICA", nullable = false)
    private Integer idCaracteristica;

    @Column(name = "NOMBRE", nullable = false, unique = true, length = 15)
    private String nombre;

    @Column(name = "UNIDAD_MEDIDA", nullable = false, length = 7)
    private String unidadMedida;

    // Constructor vacío (necesario para JPA)
    public CaracteristicaComplementaria() {
    }

    // Constructor con parámetros
    public CaracteristicaComplementaria(String nombre, 
                                        String unidadMedida) {
        this.nombre = nombre;
        this.unidadMedida = unidadMedida;
    }

    // Getters y setters
    public Integer getIdCaracteristica() {
        return idCaracteristica;
    }

    public void setIdCaracteristica(Integer idCaracteristica) {
        this.idCaracteristica = idCaracteristica;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getUnidadMedida() {
        return unidadMedida;
    }

    public void setUnidadMedida(String unidadMedida) {
        this.unidadMedida = unidadMedida;
    }
}
